dio/lib/src/adapters/browser_adapter.dart

```diff
@@ -135,16 +140,18 @@ class BrowserHttpClientAdapter implements HttpClientAdapter {
             );
             xhr.abort();
           }
-        });
+        }
+
+        xhr.upload.onprogress = startStopwatch.toJS;
       }
 
       final onSendProgress = options.onSendProgress;
       if (onSendProgress != null) {
-        xhr.upload.onProgress.listen((event) {
-          if (event.loaded != null && event.total != null) {
-            onSendProgress(event.loaded!, event.total!);
-          }
-        });
+        void onSendProgressWrap(web.ProgressEvent event) {
+          onSendProgress(event.loaded, event.total);
```

Comment:
The progress loaded and total events are now non-nullable?

Reply:
> The progress loaded and total events are now non-nullable?

Yes, they are non-nullable in `package:web`.